### PR TITLE
Single product page

### PR DESCRIPTION
**Pull request by lannisterrr**
Review First 12 files.
Added - Single Product Page
Added - Star based on rating
Added - Resposiveness

### CODE REVIEW

File: src/components/Navbar.js [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
 import { SearchBar } from 'components/input/SearchBar';
-import { useLocation } from 'react-router-dom';
+import { useLocation, useNavigate } from 'react-router-dom';
 function Navbar() {
+  const navigate = useNavigate();
   const location = useLocation();
   const { state } = useCart();
+  const { auth, setAuth } = useAuth();
+
+  console.log(auth, ' - navbar');
```
Comment (abhij1607) [inline src/components/Navbar.js:13]: can remove console.log

File: src/utils/wishlist-utils.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  try {
+    const response = await axios.post(
+      '/api/user/wishlist',
+      {
+        product,
+      },
+      config
+    );
+    dispatch({ type: 'ADD_TO_WISHLIST', payload: response.data.wishlist });
+    console.log(response.data.wishlist);
```
Comment (abhij1607) [inline src/utils/wishlist-utils.js:19]: can remove console.log

File: src/utils/wishlist-utils.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  } catch (e) {
+    console.log(e);
```
Comment (abhij1607) [inline src/utils/wishlist-utils.js:21]: can remove console.log

File: src/utils/wishlist-utils.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  try {
+    const response = await axios.delete(
+      `/api/user/wishlist/${productId}`,
+      config
+    );
+    dispatch({
+      type: 'REMOVE_FROM_WISHLIST',
+      payload: response.data.wishlist,
+    });
+    console.log(response.data.wishlist, 'delete call');
```
Comment (abhij1607) [inline src/utils/wishlist-utils.js:35]: can remove console.log

**Review by abhij1607 — COMMENTED** [review]
Overall very neat code | | Modular files  | | Got an idea on how to implement auth

File: src/App.js [hunk 5]
```diff
@@ -1,12 +1,15 @@
...
-import Cart from './pages/Cart';
-import Wishlist from './pages/Wishlist';
-import Listing from './pages/Listing';
+import { Routes, Route, useLocation, Navigate } from 'react-router-dom';
+import Home from 'pages/Home';
+import Login from 'pages/Login';
+import Cart from 'pages/Cart';
+import Wishlist from 'pages/Wishlist';
+import Listing from 'pages/Listing';
+import { ProductDetail } from 'pages/ProductDetail';
```
Comment (prachi-sahani) [inline src/App.js:10]: suggestion - you can use barrel  imports here

**Review by prachi-sahani — COMMENTED** [review]
Overall the code is well structured and clean
Good to merge

suggestion - 
- remove console.log
- remove {' '}
- format the code

**Pull request merged**